modules/netns-isolation.nix

```diff
@@ -0,0 +1,296 @@
+          connections = [ "bitcoind" ]
+          ++ ( optionals config.services.electrs.TLSProxy.enable [ "nginx" ]);
+        };
+        spark-wallet = {
+          id = 17;
+          # communicates with clightning over lightning-rpc socket
+          connections = [];
+        };
+        lightning-charge = {
+          id = 18;
+          # communicates with clightning over lightning-rpc socket
+          connections = [];
+        };
+        nanopos = {
+          id = 19;
+          connections = [ "nginx" "lightning-charge" ];
+        };
+        recurring-donations = {
+          id = 20;
+          # communicates with clightning over lightning-rpc socket
+          connections = [];
+        };
+        nginx = {
+          id = 21;
+          connections = [];
+        };
+      };
+
+      systemd.services = {
+        netns-bridge = {
+          description = "Create bridge";
+          requiredBy = [ "tor.service" ];
+          before = [ "tor.service" ];
+          script = ''
+            ${ip} link add name br0 type bridge
+            ${ip} link set br0 up
+            ${ip} addr add ${bridgeIp}/24 brd + dev br0
+            ${iptables} -w -t nat -A POSTROUTING -s 169.254.${toString cfg.addressblock}.0/24 -j MASQUERADE
+          '';
+          preStop = ''
+            ${iptables} -w -t nat -D POSTROUTING -s 169.254.${toString cfg.addressblock}.0/24 -j MASQUERADE
+            ${ip} link del br0
+          '';
+          serviceConfig = {
+            Type = "oneshot";
+            RemainAfterExit = "yes";
+          };
+        };
+
+        bitcoind-import-banlist.serviceConfig.NetworkNamespacePath = "/var/run/netns/nb-bitcoind";
+      } //
+      (let
+        makeNetnsServices = n: v: let
+          vethName = "nb-veth-${toString v.id}";
+          netnsName = "nb-${n}";
+          ipNetns = "${ip} -n ${netnsName}";
+          netnsIptables = "${ip} netns exec ${netnsName} ${config.networking.firewall.package}/bin/iptables";
+        in {
+          "${n}".serviceConfig.NetworkNamespacePath = "/var/run/netns/${netnsName}";
+
+          "netns-${n}" = rec {
+            requires = [ "netns-bridge.service" ];
+            after = [ "netns-bridge.service" ];
+            bindsTo = [ "${n}.service" ];
+            requiredBy = bindsTo;
+            before = bindsTo;
+            script = ''
+              ${ip} netns add ${netnsName}
+              ${ipNetns} link set lo up
+              ${ip} link add ${vethName} type veth peer name br-${vethName}
+              ${ip} link set ${vethName} netns ${netnsName}
+              ${ipNetns} addr add ${v.address}/24 dev ${vethName}
+              ${ip} link set br-${vethName} up
+              ${ipNetns} link set ${vethName} up
+              ${ip} link set br-${vethName} master br0
+              ${ipNetns} route add default via ${bridgeIp}
+              ${netnsIptables} -w -P INPUT DROP
+              ${netnsIptables} -w -A INPUT -s 127.0.0.1,${bridgeIp} -j ACCEPT
```

Comment:
It should be possible to allow the `ipNetns` to connect to `ipNetns`. In order to allow that I think we need to add `${ipNetns}` here and in the `OUTPUT` rule below. That allows simplifying the tests by testing connectivity from the same container. Also we can remove additional connection from clightning to spark in the tests.

Reply:
Fixed in 43a34be11cf79d252cbc0b846b12623b61c4fdbb